backend/tests/worker/test_init.py

```diff
@@ -0,0 +1,202 @@
+
+        # The pipeline has been run once, but the data wasn’t available yet
+        runs = list(db_session.execute(query).scalars())
+        assert len(runs) == 1
+        assert runs[0].status == PipelineStatus.DATA_UNAVAILABLE
+
+    # Tuesday, July 8th, at 12:10 pm
+    with time_machine.travel("2025-07-08T12:10:00+02:00"):
+        rcv_mock.return_value = PipelineResult(
+            status=PipelineStatus.SUCCESS,
+            checksum="123abc",
+        )
+
+        worker.run_pending()
+
+        # The pipeline has been run a second time and has successfully fetched the data
+        runs = list(db_session.execute(query).scalars())
+        assert len(runs) == 2
+        assert runs[1].status == PipelineStatus.SUCCESS
+        assert runs[1].checksum == "123abc"
+
+    # Tuesday, July 8th, at 12:20 pm
+    with time_machine.travel("2025-07-08T12:20:00+02:00"):
+        worker.run_pending()
+
+        # The pipeline hasn’t been run again as it was successfully run before
+        runs = list(db_session.execute(query).scalars())
+        assert len(runs) == 2
+
+    # Tuesday, July 8th, at 5:00 pm
+    with time_machine.travel("2025-07-08T17:00:00+02:00"):
+        rcv_mock.return_value = PipelineResult(
+            status=PipelineStatus.DATA_UNCHANGED,
+            checksum=None,
+        )
+
+        worker.run_pending()
+
+        # The pipeline has been run again to check for another voting session in
+        # the evening, but the data hasn’t changed
+        runs = list(db_session.execute(query).scalars())
+        assert len(runs) == 3
+        assert runs[2].status == PipelineStatus.DATA_UNCHANGED
+
+    # Tuesday, July 8th, at 5:10 pm
+    with time_machine.travel("2025-07-08T17:10:00+02:00"):
+        rcv_mock.return_value = PipelineResult(
+            status=PipelineStatus.SUCCESS,
+            checksum=None,
+        )
+
+        worker.run_pending()
+
+        # The pipeline has been run again and has successfully fetched the data
+        runs = list(db_session.execute(query).scalars())
+        assert len(runs) == 4
+        assert runs[3].status == PipelineStatus.SUCCESS
+
+    # Tuesday, July 8th, at 08:00 pm
+    with time_machine.travel("2025-07-08T20:00:00+02:00"):
+        worker.run_pending()
+
+        # The pipeline hasn’t been run again as it has already succeeded twice that day
+        runs = list(db_session.execute(query).scalars())
+        assert len(runs) == 4
+
+    # Wednesday, July 9th, at 12:00 pm
+    with time_machine.travel("2025-07-09T12:00:00+02:00"):
+        rcv_mock.return_value = PipelineResult(
+            status=PipelineStatus.SUCCESS,
+            checksum=None,
+        )
+
+        worker.run_pending()
+
+        # The pipeline has been executed again, as it’s the next day
+        runs = list(db_session.execute(query).scalars())
+        assert len(runs) == 5
+        assert runs[4].status == PipelineStatus.SUCCESS
+
+
+def test_rcv_list_notification(db_session, mocker):
```

Comment:
Can we rename this test? In the case we successfully scrape new votes, we also send a notification. This is untested, as the actual behavior of the `RCVListPipeline` is mocked, if I am not mistaken.
Hence, this tests whether the mechanism to notify us when there were no votes, but we think there might should be votes. Something like `test_rcv_list_notification_no_votes` would suffice, mainly to make it more explicit what is (and is not) tested.

Reply:
@linusha I’ve added a separate test for the entire `RCVListPipeline` which also includes the success notification.